vhost_user_backend/src/lib.rs

```diff
@@ -718,9 +720,15 @@ impl<S: VhostUserBackend> VhostUserSlaveReqHandler for VhostUserHandler<S> {
         // VHOST_USER_GET_VRING_BASE.
         self.vrings[index as usize].write().unwrap().queue.ready = false;
         if let Some(fd) = self.vrings[index as usize].read().unwrap().kick.as_ref() {
-            self.worker
-                .unregister_listener(fd.as_raw_fd(), epoll::Events::EPOLLIN, u64::from(index))
+            if let Some(epoll_fd) = self.vrings[index as usize].read().unwrap().epoll_fd {
+                unregister_listener(
+                    epoll_fd,
+                    fd.as_raw_fd(),
+                    epoll::Events::EPOLLIN,
+                    u64::from(index),
+                )
```

Comment:
If the backend is going to deal with the worker threads, it should also manage the epoll fds itself. I think this should be replaced with a backend call with index and RawFd as arguments.

On the backend, this call can also be a good hint for creating the vring worker, if it doesn't want to create them in advance.

Reply:
I agree with @slp I think the epoll fds should be managed in the backends.

Reply:
@rbradford update is done to let backends manage epoll fds.